ntbs-service/Models/PartialDate.cs

```diff
@@ -0,0 +1,59 @@
+using System;
+using System.ComponentModel.DataAnnotations.Schema;
+using ntbs_service.Models.Validations;
+
+namespace ntbs_service.Models
+{
+    [NotMapped]
+    [ValidPartialDateCanConvertToDatetime(ErrorMessage = ValidationMessages.InvalidDate)]
+    public class PartialDate
+    {
+        public string Day {get; set; }
+        public string Month {get; set; }
+        public string Year  {get; set; }
+
+        public bool IsEmpty()
+        {
+            return string.IsNullOrEmpty(Year) && string.IsNullOrEmpty(Month) && string.IsNullOrEmpty(Day);
+        }
+
+        public bool TryConvertToDateTimeRange(out DateTime? dateTimeStart, out DateTime? dateTimeEnd)
+        {
+            dateTimeStart = null;
+            dateTimeEnd = null;
+
+            try
+            {
+                if (!int.TryParse(Year, out int parsedYear)) {
+                    return false;
+                }
+                if (!int.TryParse(Month, out int parsedMonth)) {
+                    parsedMonth = 1;
+                };
+                if (!int.TryParse(Day, out int parsedDay)) {
+                    parsedDay = 1;
+                };
+
+                dateTimeStart = new DateTime(parsedYear, parsedMonth, parsedDay);
+
+                if (!string.IsNullOrEmpty(Day)) 
+                {
+                    dateTimeEnd = dateTimeStart?.AddDays(1);
+                }
+                else if(!string.IsNullOrEmpty(Month)) 
+                {
+                    dateTimeEnd = dateTimeStart?.AddMonths(1);
+                }
+                else 
+                {
+                    dateTimeEnd = dateTimeStart?.AddYears(1);
+                }
+                return true;
+            }
+            catch (ArgumentOutOfRangeException)

```

Comment:
I think it might actually be the other way round - we need to catch the exception for if the data we pass into new DateTime(x,y,z) isn't a valid date but we're only ever adding 1 day/month/year so that shouldnt through an exception

Reply:
Very fair missed that one, would make sense to wrap the bottom section then (all but the tryParse section)? 
Could hypothetically feed it year 9999 (no month/day) which would try to tick over DateTime.MaxValue, throwing the exception.